src/rules/generic-spacing.ts

```diff
@@ -3,6 +3,7 @@ import { createEslintRule } from '../utils'
 export const RULE_NAME = 'generic-spacing'
 export type MessageIds = 'genericSpacingMismatch'
 export type Options = []
+const PRESERVE_PREFIX_SPACE_BEFORE_GENERIC = new Set(['TSCallSignatureDeclaration', 'ArrowFunctionExpression', 'TSFunctionType', 'FunctionExpression'])
```

Comment:
The size has exceeded 3 after adding `FunctionExpression`, which `Set#has` would operate more efficiently than `Array#includes`.

Additionally, hoisting the Set to prevent repetitive creation of objects.